src/lib.rs

```diff
@@ -0,0 +1,83 @@
+// License: MIT Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+/*!
+# frequenz-resampling-rs
+
+This project is the rust resampler for resampling a stream of samples to a
+given interval.
+
+## Usage
+
+To resample a vector of samples to a given interval, you can use the
+`Resampler` struct.
+The construction of a resampler expects an interval (`TimeDelta`) and a
+`ResamplingFunction`.
```

Comment:
I think because constructors don't have standardized names, we should specify the function's name.  Plus you could also link to specific functions under the Usage section like this:

```suggestion
An instance of the [`Resampler`] can be created with the [`new`][Resampler::new]
method.

Raw data can be added to the resampler either through the
[`push`][Resampler::push] or [`extend`][Resampler::extend] methods, and the the
[`resample`][Resampler::resample] method resamples the data that was added to
the buffer.
```